File: day3/Remove_Duplicates.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i=1
        while(i<len(nums)):
            if(nums[i]==nums[i-1]):
                nums.pop(i)
            else:
                i=i+1
        return len(nums)

File: day0011/Remove_Duplicates.cpp
/*
code by : Vivek pandey

*/
class Solution {
public:
    string removeDuplicates(string S) {
        stack <char> st;
        int i=0;
        int n = S.size();
        while(i<n){
            if(st.empty()){
                st.push(S[i]);
            }
            
           else if(st.top()==S[i]){
                st.pop();
            }
            else{
                st.push(S[i]);
            }
            i++;
        }
        string ans;
        while(!st.empty()){
            char a = st.top();
            st.pop();
            ans.push_back(a);
        }
        reverse(ans.begin(),ans.end());
        return ans;
    }
};
